OSDOCS#12733: Update z-stream release notes for 4.15.39 (#85359)
openshift/openshift-docs: release_notes/ocp-4-15-release-notes.adoc

For any {product-title} release, always review the instructions on xref:../updating/updating_a_cluster/updating-cluster-web-console.adoc#updating-cluster-web-console[updating your cluster] properly.
====

//4.15.39
[id="ocp-4-15-39_{context}"]

> === RHSA-2024:8991 - {product-title} 4.15.39 bug fix and security update

Reviewer: ```suggestion
=== RHSA-2024:10142 - {product-title} 4.15.39 bug fix and security update
```

Issued: 26 November 2024

{product-title} release 4.15.39, which includes security updates, is now available. The list of bug fixes that are included in this update is documented in the link:https://access.redhat.com/errata/RHSA-2024:10142[RHSA-2024:10142] advisory. The RPM packages that are included in this update are provided by the link:https://access.redhat.com/errata/RHSA-2024:10145[RHSA-2024:10145] advisory.

Space precluded documenting all of the container images for this release in the advisory.

You can view the container images in this release by running the following command:

[source,terminal]
----
$ oc adm release info 4.15.39 --pullspecs
----

> [id="ocp-4-15-39-bug-fixes_{context}"]

Reviewer: Please update the release note fields in each Jira on the [Images](https://errata.devel.redhat.com/advisory/143054) advisory. I see a lot of Jiras have the Release Note Status field as "In progress".

==== Bug fixes

* Previously, the approval process for a certificate signing request (CSR) failed because the node name and internal DNS entry for a CSR did not match the case-sensitive check. With this release, an update to the approval process for CSRs avoids this check and a CSR with a matching node name and internal DNS entry does not fail the check. (link:https://issues.redhat.com/browse/OCPBUGS-44705[*OCPBUGS-44705*])

> * Previously, when the Cluster Resource Override Operator was unable to completely deploy its operand controller, it would restart the process. Each time it attempted the process, it created a new set of secrets. This resulted in a large number of secrets created in the namespace where Cluster Resource Override Operator was deployed. With this release, the fixed version correctly proceses the service account annotations and only one set of secrets is created. (link:https://issues.redhat.com/browse/OCPBUGS-44378[*OCPBUGS-44378*])

Reviewer: ```suggestion
* Previously, when the Cluster Resource Override Operator was unable to completely deploy its operand controller, the Operator would restart the process. Each time the Operator attempted the deployment process, the Operator created a new set of secrets. This resulted in a large number of secrets created in the namespace where the Cluster Resource Override Operator was deployed. With this release, the fixed version correctly proceseses the service account annotations and only one set of secrets is created. (link:https://issues.redhat.com/browse/OCPBUGS-44378[*OCPBUGS-44378*])
```

> * Previously, when the Cluster Version Operator (CVO) pod restarted while it was initializing the synchronization work, it interrupted the guard of the blocked upgrade request. The blocked request was unexpectedly accepted. With this release, the guard of the blocked upgrade request continues after the CVO restarts. (link:https://issues.redhat.com/browse/OCPBUGS-44328[*OCPBUGS-44328*])

Reviewer: ```suggestion
* Previously, when the Cluster Version Operator (CVO) pod restarted while it was initializing the synchronization work, the Operator interrupted the guard of the blocked upgrade request. The blocked request was unexpectedly accepted. With this release, the guard of the blocked upgrade request continues after the CVO restarts. (link:https://issues.redhat.com/browse/OCPBUGS-44328[*OCPBUGS-44328*])
```

> * Previously, some Microsoft Azure API calls listed available resources when using an older API that did not allow filtering by location. As a result, the API returned large amounts of data, often causing timeout errors in the installer. With this release, by updating the API version and adding a filter, the number of results returned is reduced, a timeout does not occur, and the installation continues. (link:https://issues.redhat.com/browse/OCPBUGS-43760[*OCPBUGS-43760*])

Reviewer: Some Jiras such as https://issues.redhat.com/browse/OCPBUGS-43760 do not have your wordsmithed note in the Release Note Text field. Please update the Jiras.

> * Previously, some Microsoft Azure API calls listed available resources when using an older API that did not allow filtering by location. As a result, the API returned large amounts of data, often causing timeout errors in the installer. With this release, by updating the API version and adding a filter, the number of results returned is reduced, a timeout does not occur, and the installation continues. (link:https://issues.redhat.com/browse/OCPBUGS-43760[*OCPBUGS-43760*])

Reviewer: ```suggestion
* Previously, some {azure-first} API calls listed available resources when using an older API that did not allow filtering by location. As a result, the API returned large amounts of data, often causing timeout errors in the installer. With this release, by updating the API version and adding a filter, the number of results returned is reduced, a timeout does not occur, and the installation continues. (link:https://issues.redhat.com/browse/OCPBUGS-43760[*OCPBUGS-43760*])
```

> * Previously, enabling Encapsulated Security Payload (ESP) hardware offload using IPSec on attached interfaces in Open vSwitch (OVS) interrupted connectivity because of a bug in OVS. With this release, {product-title} automatically disables ESP hardware offload on the OVS-attached interfaces, and the issue is resolved. (link:https://issues.redhat.com/browse/OCPBUGS-44240[*OCPBUGS-44240*])

Reviewer: ```suggestion
* Previously, enabling Encapsulated Security Payload (ESP) hardware offload by using IPSec on attached interfaces in Open vSwitch (OVS) interrupted connectivity because of a bug in OVS. With this release, {product-title} automatically disables ESP hardware offload on the OVS-attached interfaces so that the issue is resolved. (link:https://issues.redhat.com/browse/OCPBUGS-44240[*OCPBUGS-44240*])
```
